src/lib/interfaces/drive_interface.cpp

```diff
@@ -200,7 +200,7 @@ static std::string get_raid_hdd_vendor_model() {
 
 drive_type DriveInterface::detect_drive_type(const std::string& dev_name) {
     if (std::filesystem::is_regular_file(std::filesystem::status(dev_name))) {
-        auto device = std::filesystem::path(get_mounted_device(dev_name)).filename();
```

Comment:
I think we used std::filesystem::path() for a reason right? Will it not break that and cause regression? I believe this was done so that if one passes soft/hard link path to the device, it could pick the softlink and treat the size of the device as link size, instead of actual device?

Reply:
```
Thread 1 "test_journal_vd" hit Breakpoint 1, iomgr::DriveInterface::detect_drive_type (dev_name="/sysroot/tmp/test_journal_vdev_1") 

(gdb) p get_mounted_device(dev_name)
$1 = "/dev/mapper/atomicos-root"

(gdb) p dev_name.assign("/var/home/ubuntu/ttt")
$3 = "/var/home/ubuntu/ttt"
(gdb) p dev_name
$4 = "/var/home/ubuntu/ttt"
(gdb) p get_mounted_device(dev_name)
$5 = "/dev/mapper/atomicos-root"


storage-workstation-gtxgj:~# ln -s /sysroot/tmp/test_journal_vdev_1 ./ttt
storage-workstation-gtxgj:~# ls
2.cpp  go  ttt
storage-workstation-gtxgj:~# pwd
/var/home/ubuntu
storage-workstation-gtxgj:~# ls -rlt .
total 4
drwxrwx--- 3 ubuntu ubuntu  17 Apr  8 00:12 go
-rw-rw---- 1 ubuntu ubuntu 632 Jun  2 23:20 2.cpp
lrwxrwxrwx 1 ubuntu ubuntu  32 Jun  3 09:46 ttt -> /sysroot/tmp/test_journal_vdev_1
```

I test this using a soft link, and it works as expected

Reply:
@hkadayam can you recall for what reason we use file name instead of the absolute path to get the the major and minor number of a device